### PR TITLE
Add nftables test module to test it's support on sles 15

### PR DESCRIPTION
**Pull request by Zaoliang**
see  https://progress.opensuse.org/issues/88723
and https://jira.suse.com/browse/SLE-16460
schedule this module in extra_test_texmode for sles 15
verification:
http://openqa.suse.de/tests/5642240#step/nftables/55615

### CODE REVIEW

File: tests/console/nftables.pm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    $self->select_serial_terminal;
+
+    # check nftables, stop iptables and start firewalld by need
+    my $logs = "/tmp/log";
+
+    #check that nftables has been already installed
+    assert_script_run('echo -e "____ nftables version and rules ____\n\n" > ' . $logs);
+    assert_script_run("rpm -qa nftables |& tee -a $logs");
+    assert_script_run("if systemctl is-active -q iptables; then systemctl stop iptables; fi");
+    assert_script_run("if ! systemctl is-active -q firewalld; then systemctl start firewalld; fi");
```
Comment (asdil12) [inline tests/console/nftables.pm:31]: Why the `if` - you could simply always stop and start that service.

Comment (Zaoliang) [reply tests/console/nftables.pm]: yes, just restart firewalld now:  http://10.160.64.152/tests/637#step/nftables/20

File: tests/console/nftables.pm [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # check nftables, stop iptables and start firewalld by need
+    my $logs = "/tmp/log";
+
+    #check that nftables has been already installed
+    assert_script_run('echo -e "____ nftables version and rules ____\n\n" > ' . $logs);
+    assert_script_run("rpm -qa nftables |& tee -a $logs");
+    assert_script_run("if systemctl is-active -q iptables; then systemctl stop iptables; fi");
```
Comment (asdil12) [inline tests/console/nftables.pm:30]: But why the stop before the restart?

File: tests/console/nftables.pm [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    assert_script_run("systemctl restart firewalld");
```
Comment (asdil12) [inline tests/console/nftables.pm:31]: Please use the `systemctl` perl function:

```suggestion
    systemctl("restart firewalld");
```

File: tests/console/nftables.pm (hunk 2 above)
Comment (Zaoliang) [reply tests/console/nftables.pm:30]: this is about iptables :)

**Review by asdil12 — APPROVED** [review]

**Comment by DimStar77** [comment]
> see https://progress.opensuse.org/issues/88723

Top-secret Jira ticket - what happened to 'open open source company'

> and https://jira.suse.com/browse/SLE-16460
> schedule this module in extra_test_texmode for sles 15
> verification:
> http://openqa.suse.de/tests/5642240#step/nftables/55615

No test for Tumbleweed?
Needless to say: https://openqa.opensuse.org/tests/1665423#step/nftables/20 = FAIL

**Comment by Zaoliang** [comment]
I didn't know that the yaml schedule is used for TW too :)

Well, the request in ticket/jira was first of all for sles 15. 
Sorry, if you cannot access jira ticket, I can understand this, however there is no secret about nftables support. In poo ticket is clear metioned. 
I'll check this issue for TW as well: https://progress.opensuse.org/issues/90041
The previous module firewalld failed as well, strange?

File: tests/console/nftables.pm [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    # open tcp ports: sshd (22), httpd (80)
+    tcp dport {ssh, http} accept
+  }
+}
+
+EOF
+true');
+
+    assert_script_run("nft -f firewall");
```
Comment (foursixnine) [inline tests/console/nftables.pm:101]: You are only checking that this call worked, there's no verification if the rule was actually loaded

File: tests/console/nftables.pm [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+EOF
+true');
+
+    # run and check customer rule, add firewall rule and restart firewalld
+    assert_script_run("nft flush ruleset");
+    assert_script_run("nft -f tcp-mss-clamp");
+    assert_script_run("nft list chain inet firewalld filter_FWDO_public_allow");
+    assert_script_run("nft list tables");
```
Comment (foursixnine) [inline tests/console/nftables.pm:52]: These tests depend on a set of rules already being present, but none of them are actually checking that the output is what should be expected

**Review by foursixnine — COMMENTED** [review]
@Zaoliang See my comments on this PR, and please adress them on # #12170 (cc @asdil12)

File: tests/console/nftables.pm [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # check nftables, stop iptables and start firewalld by need
+    my $logs = "/tmp/log";
+
+    #check that nftables has been already installed
+    assert_script_run('echo -e "____ nftables version and rules ____\n\n" > ' . $logs);
+    assert_script_run("rpm -qa nftables |& tee -a $logs");
```
Comment (foursixnine) [inline tests/console/nftables.pm:29]: I guess you're tring to get the version here, please use: zypper_call('info nftables'); instead

File: tests/console/nftables.pm (hunk 4 above)
Comment (Zaoliang) [reply tests/console/nftables.pm:101]: the rules from 'firewall' got checked later by 'nft list output'

File: tests/console/nftables.pm (hunk 6 above)
Comment (Zaoliang) [reply tests/console/nftables.pm:29]: actually this is almost the same what I want to check: build information should be there for logs only if something goes wrong and we know which version.

File: tests/console/nftables.pm (hunk 4 above)
Comment (Zaoliang) [reply tests/console/nftables.pm:101]: please see:
assert_script_run("firewall-cmd --list-services --zone=public |& tee -a $logs");

Comment (foursixnine) [reply tests/console/nftables.pm:101]: that requires the reviewer to check the logs to know if the output matches the expected string... please use grep and search for the expected strings.

File: tests/console/nftables.pm (hunk 6 above)
Comment (foursixnine) [reply tests/console/nftables.pm:29]: Please use the suggested `zypper_call('info nftables');`

File: tests/console/nftables.pm (hunk 4 above)
Comment (Zaoliang) [reply tests/console/nftables.pm:101]: okay, will add grep then. 
I'll ask you to review changes in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12170 later, thanks!

**Pull request merged**